app/src/main/java/com/kotlin/readyplayerme/WebViewActivity.kt

```diff
@@ -150,47 +203,78 @@ class WebViewActivity : AppCompatActivity() {
                         return null;
                     }
                 }
-    
-                window.postMessage(
-                    JSON.stringify({
-                        target: 'readyplayerme',
-                        type: 'subscribe',
-                        eventName: 'v1.**'
-                    }),
-                    '*'
-                );
 
                 window.removeEventListener('message', subscribe);
                 window.addEventListener('message', subscribe);
             """.trimIndent(), null)
         }
     }
 
-    @SuppressLint("SetJavaScriptEnabled", "JavascriptInterface")
-    private fun setUpWebView(isCreateNew: Boolean) {
-        Log.i("WEBVIEWACTIVITY", "onCreate: isCreateNew $isCreateNew")
-        with(binding.webview.settings){
-            javaScriptEnabled = true
-            cacheMode = WebSettings.LOAD_DEFAULT
-            databaseEnabled = true
-            domStorageEnabled = true
-            allowFileAccess = true
+    private fun handleWebMessage(webMessage: WebMessage) {
 
+        when (webMessage.eventName) {
+            WebViewInterface.WebViewEvents.USER_SET -> {
+                val userId = webMessage.data.getStringProperty(ID_KEY)
+                if (userId != null) {
+                    callback?.onOnUserSet(userId)
+                }
+            }
+            WebViewInterface.WebViewEvents.USER_UPDATED -> {
+                val userId = webMessage.data.getStringProperty(ID_KEY)
+                if (userId != null) {
+                    callback?.onOnUserUpdated(userId)
+                }
+            }
+            WebViewInterface.WebViewEvents.USER_AUTHORIZED -> {
+                val userId = webMessage.data.getStringProperty(ID_KEY)
+                if (userId != null) {
+                    callback?.onOnUserAuthorized(userId)
+                }
+            }
+            WebViewInterface.WebViewEvents.ASSET_UNLOCK -> {
+                val userId = webMessage.data.getStringProperty(ID_KEY)
+                val assetId = webMessage.data.getStringProperty(ASSET_ID_KEY)
+                if(userId != null && assetId != null){
```

Comment:
I wonder if failing silently is the best approach here. I'm just thinking that if this is a sample for people to use, would it not be better to throw and exception if the data provided on the postMessage doesn't match up to what is expected? This way it we won't get support messages like 'hey, I'm not getting X event' because it's failing silently and the developers doesn't know.

IMO it would be better to fail fast and throw an exception. Then if a partner notices an issue because something is wrong with the webview events  they can either modify the code to fail silently themselves, or get in touch with us with a much more helpful message about what is wrong / what property is suddenly missing?